src/backend/dx12/src/device.rs

```diff
@@ -146,7 +147,11 @@ pub(crate) fn compile_shader(
             ptr::null_mut(),
             entry.as_ptr() as *const _,
             full_stage.as_ptr() as *const i8,
-            1,
+            d3dcompiler::D3DCOMPILE_DEBUG | if features.contains(hal::Features::UNSIZED_DESCRIPTOR_ARRAY) {
```

Comment:
This changes the default value from 1 to 0.
Let's move this out and add a bit of logic here, something like:
```rust
let mut compile_flags = d3dcompiler::D3DCOMPILE_ENABLE_STRICTNESS;
if cfg!(debug_assertions) {
  compile_flags |= d3dcompiler::D3DCOMPILE_DEBUG;
}
if features.contains(hal::Features::UNSIZED_DESCRIPTOR_ARRAY) {
  compile_flags |= d3dcompiler::D3DCOMPILE_ENABLE_UNBOUNDED_DESCRIPTOR_TABLES;
}
```

Reply:
`D3DCOMPILE_DEBUG` is defined as `1 << 0` so that would seem to be the value of 1 it was set to before. I agree with the general refactor.